src/analyses/base.ml

```diff
@@ -1251,13 +1252,13 @@ struct
     | Q.EvalLength e -> begin
         match eval_rv_address (Analyses.ask_of_ctx ctx) ctx.global ctx.local e with
         | `Address a ->
-          let slen = List.map String.length (AD.to_string a) in
           let lenOf = function
             | TArray (_, l, _) -> (try Some (lenOfArray l) with LenOfArray -> None)
             | _ -> None
           in
-          let alen = List.filter_map (fun v -> lenOf v.vtype) (AD.to_var_may a) in
-          let d = List.fold_left ID.join (ID.bot_of (Cilfacade.ptrdiff_ikind ())) (List.map (ID.of_int (Cilfacade.ptrdiff_ikind ()) %BI.of_int) (slen @ alen)) in
+          let d = Iter.((Iter.map String.length (Iter.of_list @@ AD.to_string a)) <+> (Iter.filter_map (fun v -> lenOf v.vtype) (Iter.of_list @@ AD.to_var_may a)) (*<- equal to slen @ alen *)
```

Comment:
If you already have `Iter.(...)` outside, you do not need it inside again.

Reply:
Right, that somehow slipped my mind. Should I push a fix for that even though the code deadline has passed?